infrastructure/server/network_security_groups.tf

```diff
@@ -0,0 +1,121 @@
+resource "aws_security_group" "fam_app_sg" {
+    name = "fam_app_sg"
+    description = "FAM custom security group for application tier (lambdas)."
+    vpc_id = data.aws_vpc.selected.id
+    revoke_rules_on_delete = true
+
+    tags = {
+        Name = "fam_app_sg"
+        managed-by = "terraform"
+    }
+
+    ingress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["10.10.32.0/20", "10.10.128.0/20"]
+        description = "Central VPC Traffic Inbound from Web subnets"
+    }
+
+    ingress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["10.10.0.0/19", "10.10.96.0/19"]
+        description = "Central VPC Traffic Inbound from App subnets"
+    }
+
+    ingress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["10.10.64.0/21", "10.10.72.0/21"]
+        description = "Central VPC Traffic Inbound from Mgmt subnets"
+    }
+
+    egress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["0.0.0.0/0"]
+        description = "Allow All Outbound Traffic"
+    }
+
+}
+
+resource "aws_security_group" "fam_data_sg" {
+    name = "fam_data_sg"
+    description = "FAM custom security group for data tier."
+    vpc_id = data.aws_vpc.selected.id
+    revoke_rules_on_delete = true
+    tags = {
+        Name = "fam_data_sg"
+        managed-by = "terraform"
+    }
+}
```

Comment:
Hi Basil, just curious... and probably you did this for a reason...
Why does ingress/egress rules were defined in the same block in "**fam_app_sg**" resource but rules are being separated from "**fam_data_sg**" resource?
Was just two different styles or maybe some difficulties you found?

Reply:
the separate rules is the preferred style, but I didn't discover that initially. Plus for fam_data_sg I needed a rule to reference the same security group, not sure if that's possible with the inline format.